bbc/simorgh
# A11y update for Post [Time, Breaking, Headline]

Resolves [WSTEAM1-718](https://jira.dev.bbc.co.uk/browse/WSTEAM1-718) - A11y update for Post [Time, Breaking, Headline]

(Stems from A11y update required for A11y A/C following the merger of the [WSTEAM1-621](https://github.com/bbc/simorgh/pull/11065) & [WSTEAM1-449](https://github.com/bbc/simorgh/pull/11089) and [WSTEAM1-448](https://github.com/bbc/simorgh/pull/11078) into latest)

Overall changes
======
Suggestioned changes to the A11y A/Cs are met for the Posts component.

See in the POST A11y A/C's [here](https://paper.dropbox.com/doc/Live-page-Accessibility-Acceptance-Criteria--CAT5uRt5yiqY2K7Hkz5k7rZOAg-KZQODmAf9tvodUlxsAVnA#:h2=POSTS).

Code changes
======

A11y
- Move Breaking & Timestamp label inside the H3, to be read by screenreaders in a single gesture
- Add visually hidden comma/ spacing to add gap when css is disabled.
- Update styling to use 'spacings' rather then pixelsToRem() where possible.
- Change outer div of post from Div to Article. This is what is says on the Screen Reader UX.
- Change date format to D MM YYYY, so it appears as  ‘9 May 2023’ and not ‘09 May 2023’

UX
- Update styling to handle the placement of the Breaking News banner inside of the H3 element.
- Add/ Update transparent borders used in High Contrast Mode. Adjust padding/ styles as necessary.
- Fix bug with padding at the bottom of the Bulleted List (as Catharine pointed out this out - should now be 16px padding on all screen sizes)
- Update padding-inline-start on Unordered Lists to match UX designs. 
- Stacks 'timestamp' and 'breaking' on any screen size where necessary to handle longer translations.

Also
- Fix pixel declaration bug in legacy component styles and update snapshots.

Testing
======
1. Visit the[ story permalink ](https://breaking-label-a11y-suggestions--5d28eb3fe163f6002046d6fa.chromatic.com/?path=/story/pages-live-page--example). Here you can check the HTML via the inspector, test with a screenreader and test in High contrast mode.
3. Run the nextJS app locally by pulling down this branch, cd into ws-nextjs-app, run yarn dev, and visit http://localhost:7081/pidgin/live/c7p765ynk9qt.

Helpful Links
======
[UX Designs](https://www.figma.com/file/kBQdRE6Tc5XZ9L2HejAUgD/Live-mvp-post-stream---handover?type=design&node-id=355%3A36624&mode=design&t=aGsWpq7x1ETiVYFy-1)
[Screen Reader UX](https://www.figma.com/file/kBQdRE6Tc5XZ9L2HejAUgD/Live-mvp-post-stream---handover?type=design&node-id=502-61213&mode=design&t=VU7JD8ZfeYHZX28f-0)
[A11y Spec](https://paper.dropbox.com/doc/KZQODmAf9tvodUlxsAVnA?cl=en_US)

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)
